integration/kotlinx-coroutines-play-services/src/Tasks.kt

```diff
@@ -91,17 +136,28 @@ public suspend fun <T> Task<T>.await(): T {
         } else {
             throw e
         }
+    } else if (cancellationTokenSource.token.isCancellationRequested) {
+        // The task hasn't completed, yet cancellation was already requested.
+        // Interpret this by throwing immediately (no way to cancel the task)
+        throw CancellationException("Cancellation was already requested")
```

Comment:
I added handling here (and for `asDeferred`) for the case where the `CancellationTokenSource` is unrelated to the `Task`, in which case we have 3 things to coordinate:
- The `Task`'s cancellation, which we can listen for but can't directly cause
- The `CancellationTokenSource`'s cancellation, which we can listen for and cause
- The coroutine's cancellation, which we can listen for and cause

Reply:
In the latest commit, I removed this and documented that an unrelated `CancellationTokenSource` is not supported.